### PR TITLE
[NIFI-7705] update frontend deps

### PR DESCRIPTION
**Pull request by scottyaslan**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Updates frontend dependencies

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json [hunk 1]
```diff
@@ -144,20 +144,20 @@
       "resolved": "https://registry.npmjs.org/d3-brush/-/d3-brush-1.0.4.tgz",
       "integrity": "sha1-AMLyOAGfJPbAoZSibUGhUw/+e8Q=",
       "requires": {
-        "d3-dispatch": "1.0.3",
-        "d3-drag": "1.2.1",
-        "d3-interpolate": "1.1.6",
-        "d3-selection": "1.3.0",
-        "d3-transition": "1.1.1"
+        "d3-dispatch": "1",
```
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json:147]: This change may appear strange but what it is saying is that "d3-brush" version 1.0.4 requires any minor or patch release of "d3-dispatch" version 1. This however does not change the version of "d3-dispatch" that is installed. That is defined here: https://github.com/apache/nifi/pull/4449/files#diff-9a4616626c8b30875e090d2d589ce665R173.

**Comment by sardell** [comment]
Reviewing now.

**Comment by rfellows** [comment]
will review

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json (hunk 1 above)
Comment (sardell) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json:147]: I can confirm this is how their new versioning approach works and that `requires` is just used for version mismatch warnings. The top-level dependencies determines what is installed.

Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json:147]: @scottyaslan @sardell Thanks for the analysis here. If I've understood these comments correctly, they explain the behavior for a direct dependency of NiFi. They even touch on what happens is a conflict in versioned between a direct dependency of NiFi that is also a transitive dependency. 

But what is the behavior for something that is only a transitive dependency and there is no direct dependency to compare against? If that `requires` section does not contain a specific version, what ensures that when newer versions become available they aren't used? Looking to confirm that we can claim that we have repeatable builds.

Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/frontend/package-lock.json:147]: @scottyaslan @sardell I think I misunderstood the top-level dependency comment. It looks like all dependencies (transitive or not) will be 'top-level' in the package-lock.json. These will have explicit versions and will ensure repeatable builds assuming we use it in our builds.

**Comment by sardell** [comment]
LGTM. +1 (non-binding)

**Comment by rfellows** [comment]
It looks like upgrading slick-grid may have had some adverse effects on some of the usages of it...

In particular:
* Parameter tab in Parameter Context dialog
parameters

* Variables
variables


I checked around for other usages of the grid and didn't see any other issues.

**Comment by scottyaslan** [comment]
Thanks @rfellows for the thorough review. I have updated this PR to address the table issues you found.

**Pull request closed without merging**